### PR TITLE
Area schedule

### PR DESCRIPTION
**Pull request by ksobon**
### Purpose

This PR addresses this issue: https://github.com/DynamoDS/DynamoRevit/issues/1704

image

### Declarations

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.

### Reviewers

@mjkkirschner @smangarole, @jnealb

### CODE REVIEW

**Comment by Racel** [comment]
Added to our backlog. But, this requires further review before running tests. We will let you know if changes need to be made.

File: src/Libraries/RevitNodes/Elements/Views/ScheduleView.cs [hunk 1]
```diff
@@ -63,6 +63,14 @@ private ScheduleView(Category category, string name, ScheduleType type)
             SafeInit(() => InitScheduleView(category, name, type));
         }
 
+        /// <summary>
+        /// Private constructor
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Views/ScheduleView.cs]: can you note what this constructor is for vs the others? This one seems very specific.

File: src/Libraries/RevitNodes/Elements/Views/ScheduleView.cs [hunk 2]
```diff
@@ -87,17 +95,33 @@ private void InitScheduleView(Category category, string name, ScheduleType type)
...
-                vs = CreateViewSchedule(category, name, type);
-            }
+            InternalSetScheduleView(vs);
+
+            TransactionManager.Instance.TransactionTaskDone();
+            ElementBinder.CleanupAndSetElementForTrace(doc, InternalElement);
+        }
+
+        /// <summary>
+        /// Initialize a ViewSchedule Element.
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Views/ScheduleView.cs]: can you describe these overloads vs each other.

File: src/Libraries/RevitNodes/Elements/Views/ScheduleView.cs [hunk 3]
```diff
@@ -162,36 +201,56 @@ public static ScheduleView CreateSchedule(Category category, string name, string
...
+                throw new ArgumentNullException(Properties.Resources.NameArgumentException);
             }
             if (scheduleType == null)
             {
-                throw new System.ArgumentNullException(Properties.Resources.ScheduleTypeArgumentException);
+                throw new ArgumentNullException(Properties.Resources.ScheduleTypeArgumentException);
             }
 
-            ScheduleType t = (ScheduleType)System.Enum.Parse(typeof(ScheduleType), scheduleType);
+            var t = (ScheduleType)Enum.Parse(typeof(ScheduleType), scheduleType);
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Views/ScheduleView.cs]: hmm, I know this was fine before, but what do you think about wrapping this parse in a try catch explicitly with some kind of unsupported schedule type message?

**Comment by mjkkirschner** [comment]
@ksobon this looks solid, I have a few questions, and we'll need to run the tests locally if you're still having issues there.

Also, theres some small merge conflicts.

**Comment by Racel** [comment]
@ksobon - Can you let us know what the state of this PR is? If you do not have time to address the issues mentioned by @mjkkirschner above and the merge conflicts, can you please close this PR and re-open at a later date?

**Pull request closed without merging**